src/meta/duplication/meta_duplication_service.cpp

```diff
@@ -336,49 +336,120 @@ void meta_duplication_service::create_follower_app_for_duplication(
+            if (update_err == ERR_OK) {
+                blob value = dup->to_json_blob();
+                // Note: this function is `async`, it may not be persisted completed
+                // after executing, now using `_is_altering` to judge whether `updating` or
+                // `completed`, if `_is_altering`, dup->alter_status() will return `ERR_BUSY`
+                _meta_svc->get_meta_storage()->set_data(std::string(dup->store_path),
+                                                        std::move(value),
+                                                        [=]() { dup->persist_status(); });
+            } else {
+                derror_f("created follower app[{}.{}] to trigger duplicate checkpoint failed: "
+                         "duplication_status = {}, create_err = {}, update_err = {}",
+                         get_current_cluster_name(),
+                         dup->app_name,
+                         duplication_status_to_string(dup->status()),
+                         create_err.to_string(),
+                         update_err.to_string());
+            }
+        });
+}
+
+void meta_duplication_service::check_follower_app_if_create_completed(
+    const std::shared_ptr<duplication_info> &dup)
+{
+    rpc_address meta_servers;
+    meta_servers.assign_group(dup->follower_cluster_name.c_str());
+    meta_servers.group_address()->add_list(dup->follower_cluster_metas);
+
+    configuration_query_by_index_request meta_config_request;
+    meta_config_request.app_name = dup->app_name;
+
+    dsn::message_ex *msg = dsn::message_ex::create_request(RPC_CM_QUERY_PARTITION_CONFIG_BY_INDEX);
+    dsn::marshall(msg, meta_config_request);
     rpc::call(meta_servers,
               msg,
               _meta_svc->tracker(),
-              [=](error_code err, configuration_create_app_response &&resp) mutable {
-                  FAIL_POINT_INJECT_NOT_RETURN_F("update_app_request_ok",
-                                                 [&](string_view s) -> void { err = ERR_OK; });
-                  error_code create_err = err == ERR_OK ? resp.err : err;
+              [=](error_code err, configuration_query_by_index_response &&resp) mutable {
+                  FAIL_POINT_INJECT_NOT_RETURN_F("create_app_ok", [&](string_view s) -> void {
+                      err = ERR_OK;
+                      int count = dup->partition_count;
+                      while (count-- > 0) {
+                          partition_configuration p;
+                          p.primary = rpc_address("127.0.0.1", 34801);
+                          p.secondaries.emplace_back(rpc_address("127.0.0.1", 34801));
+                          p.secondaries.emplace_back(rpc_address("127.0.0.1", 34801));
+                          resp.partitions.emplace_back(p);
+                      }
+                  });
+
+                  error_code query_err = err == ERR_OK ? resp.err : err;
+                  if (query_err == ERR_OK) {
```

Comment:
How about adding some comments for this if condition? This condition is a lot bit long.